app/components/move_message.py

```diff
@@ -34,6 +35,7 @@ async def select_channel(
             return
         self._used = True
         channel = await bot.fetch_channel(sel.values[0].id)
+        assert isinstance(channel, GuildTextChannel)
```

Comment:
`channel`'s type narrowing seems wrong here:
1. too broad: the channel selector doesn't allow choosing a `ForumChannel` (applies to `utils.py:76` too)
2. too narrow: it is later cast to `TextChannel`, while in reality it can also be a `Thread` (L61, L67)